deploy/aws/java11Exec/src/main/java/poseidon/App.java

```diff
@@ -0,0 +1,136 @@
+class AwsFunctionRequest {
+    String[] cmd;
+    Map<String, String> files;
+}
+
+// WebSocketMessageType are the types of messages that are being send back over the WebSocket connection.
+enum WebSocketMessageType {
+    WebSocketOutputStdout("stdout"),
+    WebSocketOutputStderr("stderr"),
+    WebSocketOutputError("error"),
+    WebSocketExit("exit");
+
+    private final String typeName;
+
+    WebSocketMessageType(String name) {
+        this.typeName = name;
+    }
+
+    public String toString() {
+        return typeName;
+    }
+}
+
+// WebSocketMessage is the object that is beeing send over the WebSocket connection.
+class WebSocketMessage {
+    WebSocketMessageType type;
+    String data;
+    int exitCode;
+}
+
+/**
+ * Handler for requests to Lambda function.
+ * This Lambda function executes the passed command with the provided files in an isolated Java environment.
+ */
+public class App implements RequestHandler<APIGatewayV2WebSocketEvent, APIGatewayProxyResponseEvent> {
+
+    // gson helps parse the json objects.
+    private static final Gson gson = new Gson();
+
+    // gwClient is used to send messages back via the WebSocket connection.
+    private AmazonApiGatewayManagementApi gwClient;
+
+    // connectionID helps to identify the WebSocket connection that has called this function.
+    private String connectionID;
+
+    public static final String disableOutputHeaderKey = "disableOutput";
+
+    // disableOutput: If set to true, no output will be sent over the WebSocket connection.
+    private boolean disableOutput = false;
+
+    public APIGatewayProxyResponseEvent handleRequest(final APIGatewayV2WebSocketEvent input, final Context context) {
+        APIGatewayV2WebSocketEvent.RequestContext ctx = input.getRequestContext();
+        this.gwClient = AmazonApiGatewayManagementApiClientBuilder.standard()
+                .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration("https://" + ctx.getDomainName() + "/" + ctx.getStage(), Regions.EU_CENTRAL_1.getName()))
+                .build();
+        this.connectionID = ctx.getConnectionId();
+        this.disableOutput = input.getHeaders() != null && input.getHeaders().containsKey(disableOutputHeaderKey) && Boolean.parseBoolean(input.getHeaders().get(disableOutputHeaderKey));
+        AwsFunctionRequest execution = gson.fromJson(input.getBody(), AwsFunctionRequest.class);
+
+        try {
+            File workingDirectory = this.writeFS(execution.files);
+
+            ProcessBuilder pb = new ProcessBuilder(execution.cmd).redirectErrorStream(true);
+            pb.directory(workingDirectory);
+            Process p = pb.start();
+            InputStream stdout = p.getInputStream(), stderr = p.getErrorStream();
+            this.forwardOutput(p, stdout, stderr);
+            p.destroy();
+            return new APIGatewayProxyResponseEvent().withStatusCode(200);
+        } catch (Exception e) {
```

Comment:
Catching any Exception here is very generic but probably fine to "forward" it to the WebSocket consumer